Geometry/TrackerGeometryBuilder/src/TrackerGeometry.cc

```diff
@@ -254,3 +274,67 @@ TrackerGeometry::detIds()   const
 {
   return theDetIds;
 }
+void TrackerGeometry::fillTestMap(const GeometricDet* gd) {
+    
+  std::string temp = gd->name();
+  std::string name = temp.substr(temp.find(":")+1); 
+  DetId detid = gd->geographicalId();
+  float thickness = gd->bounds()->thickness();
+  std::string nameTag;  
+  TrackerGeometry::ModuleType mtype = moduleType(name);
+  if (theDetTypetList.size() == 0) {
+    theDetTypetList.push_back({std::make_tuple(detid, mtype, thickness)});
+  } else {
+    auto  & t = (*(theDetTypetList.end()-1));
+    if (std::get<1>(t) != mtype) theDetTypetList.push_back({std::make_tuple(detid, mtype, thickness)});
+    else {
+      if  ( detid > std::get<0>(t) ) std::get<0>(t) = detid;
+    }
+  }
+}
+
+TrackerGeometry::ModuleType TrackerGeometry::getDetectorType(DetId detid) const {
+  for (auto iVal : theDetTypetList) {
+    DetId detid_max = std::get<0>(iVal);
+    TrackerGeometry::ModuleType mtype =  std::get<1>(iVal);     
+    if (detid.rawId() <=  detid_max.rawId()) return mtype;
+  }
+  return TrackerGeometry::ModuleType::UNKNOWN;
+}
+float TrackerGeometry::getDetectorThickness(DetId detid) const {
+  for (auto iVal : theDetTypetList) {
+    DetId detid_max = std::get<0>(iVal);
+    if (detid.rawId() <=  detid_max.rawId()) 
+      return std::get<2>(iVal);
+  }
+  return -1.0;
+}
+
+TrackerGeometry::ModuleType TrackerGeometry::moduleType(std::string name) const {
```

Comment:
Discussing with Suchandra, the use of the templated method makes the code more complex and less readable because the mapping between module name and module type is not 1-to-1. I think that the present solution is ok.